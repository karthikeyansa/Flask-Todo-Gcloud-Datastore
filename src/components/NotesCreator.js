import React from "react";
import {API} from "../Apiservice";


var task_counter = {counter: 1};
var task_array = [1];

function NotesCreator(props){
    const initalTaskAdder = (event) => {
        if(event.keyCode === 13 && !event.shiftKey){
            event.preventDefault();
            taskadder();
        }
        else if(event.keyCode === 13 && event.shiftKey){
            event.target.style.cssText = "height:50px;"
            return false;
        }
    }

    const logoutUser = () => {
        window.location.href = "/logout";
    }

    const profileViewer = () => {
        window.location.href = "/profile";
    }

    const addedNewNote = async() =>{
        let submit = document.getElementById("submit");
        submit.setAttribute("disabled", true);
        let response = await AddNote();
        props.newAddednote({content: response.new_note.content, id: response.new_note.id,  tasks: response.new_note.tasks});
    }
    
    return (
        <React.Fragment>
            <h1 style={{textAlign:"center"}}>Welcome to Notes</h1>
            <button onClick={profileViewer} className="primary-btn" 
                    style={{right:"15%", position:"absolute"}}>Profile</button>
            <button onClick={logoutUser} className="danger-btn" 
                    style={{right:"10%", position:"absolute"}}>Logout</button>

            <div style={{display:"block", height:"25px"}}></div>
            <div className="alerts" id="alerts"></div>
            <div style={{display:"block", height:"40px"}}></div>

            <h5 style={{textAlign:"center"}}>Add here.</h5>
            <div className="maingrid" id="maingrid">
                <input type="text" className="input" name="content" id="content"
                       placeholder="Title" required/>&nbsp;
                <div id="inputTask">
                    <textarea type="text" className="textarea" 
                              placeholder="List item" id="task1"
                              onKeyDown={initalTaskAdder}></textarea>
                    <span id="span1" className="spanClose" onClick={() => inputAndSpanRemover(1)}>❌</span>
                </div>
                <br />
                <button className="primary-btn" id="submit" title="click to add to notes" 
                        onClick={addedNewNote}>➕Add to notes</button>
            </div>
        </React.Fragment>
    )
}

function taskadder() {
    var inputdiv = document.getElementById("inputTask")
    task_counter['counter'] += 1
    task_array.push(task_counter['counter'])
    if (task_array.length >= 5) {
        let maingrid = document.getElementById("maingrid");
        maingrid.style.height = "310px";
        maingrid.scrollTop = maingrid.scrollHeight;
    }

    let inputBox = document.createElement("textarea");
    inputBox.className = "textarea";
    inputBox.setAttribute("id", `task${task_counter['counter']}`);
    inputBox.placeholder = "List item";

    let spanClose = document.createElement("span");
    spanClose.setAttribute("id", `span${task_counter['counter']}`);
    spanClose.setAttribute("class", "spanClose")
    spanClose.addEventListener("click", function(event){
        let id = event.target.id;
        let value = id.replace("span", "");
        inputAndSpanRemover(value);
        
    })
    spanClose.textContent = "❌";

    inputdiv.appendChild(inputBox);
    inputdiv.appendChild(spanClose);

    let input = document.getElementById(`task${task_counter['counter']}`);
    input.focus();
    input.select();
    input.addEventListener("keydown", function (event) {
        if (event.keyCode === 13 && !event.shiftKey) {
            event.preventDefault();
            taskadder();
        }
        else if (event.keyCode === 13 && event.shiftKey) {
            event.target.style.cssText = "height:50px;"
            return false;
        }
    })
}

function inputAndSpanRemover(id){
    if (task_array.length === 1) {
        let inputTasks = document.getElementById("inputTask");
        let textarea = document.createElement("textarea");
        textarea.className = "textarea";
        textarea.placeholder = 'List item';
        textarea.setAttribute("id", 'task1');

        let spanClose = document.createElement("span");
        spanClose.setAttribute("id", "span1");
        spanClose.className = "spanClose";
        spanClose.textContent = "❌";
        spanClose.addEventListener("click", function(event){
            inputAndSpanRemover(1);            
        })

        textarea.addEventListener("keydown", function (event) {
            if (event.keyCode === 13 && !event.shiftKey) {
                event.preventDefault();
                taskadder();
            }
            else if (event.keyCode === 13 && event.shiftKey) {
                this.value += "\n";
                return false;
            }
        })
        inputTasks.innerHTML = ""
        inputTasks.appendChild(textarea);
        inputTasks.appendChild(spanClose);
    }
    else{
        let inputBox = document.getElementById(`task${id}`);
        let spanClose = document.getElementById(`span${id}`);
        inputBox.remove();
        spanClose.remove();
        let idNumber = Number(id);

        let indexToRemove = task_array.indexOf(idNumber);
        task_array.splice(indexToRemove, 1);
    }
    if (task_array.length < 5) {
        let maingrid = document.getElementById("maingrid");
        maingrid.style.height = "auto";
    }
} 

async function AddNote() {
    try {
        let note = document.getElementById("content").value;
        let tasks = [];
        task_array.forEach((element) => {
            let task_content = document.getElementById(`task${element}`).value;
            if (task_content !== "") {
                tasks.push({ "description": task_content });
            }
        })
        if(note !== ""){
            let response = await API.AddNoteAPI({"content": note, "tasks": tasks});
            if(response.success === true){
                let title = document.getElementById('content');
                title.value = "";
                let inputTasks = document.getElementById("inputTask");
                task_array.splice(0, task_array.length);
                task_counter['counter'] = 1;
                task_array = [1];

                let textarea = document.createElement("textarea");
                textarea.className = "textarea";
                textarea.placeholder = 'List item';
                textarea.setAttribute("id", 'task1');

                let spanClose = document.createElement("span");
                spanClose.setAttribute("id", "span1");
                spanClose.className = "spanClose";
                spanClose.textContent = "❌";
                spanClose.addEventListener("click", function(event){
                    inputAndSpanRemover(1);            
                })

                textarea.addEventListener("keydown", function (event) {
                    if (event.keyCode === 13 && !event.shiftKey) {
                        event.preventDefault();
                        taskadder();
                    }
                    else if (event.keyCode === 13 && event.shiftKey) {
                        this.value += "\n";
                        return false;
                    }
                })
                inputTasks.innerHTML = ""
                inputTasks.appendChild(textarea);
                inputTasks.appendChild(spanClose);
                let maingrid = document.getElementById("maingrid");
                maingrid.style.height = "auto";
            }
            else if(response.success === false){
                let alerts = document.getElementById("alerts");
                alerts.style.display = "block";
                alerts.style.backgroundColor = "red";
                alerts.innerText = "Token missing kindly login again.";
                setTimeout(function(){
                    alerts.innerText = "";
                    alerts.style.backgroundColor = "white";
                    alerts.style.display = "none";
                }, 3000)
            }
            return response;
        }
        else if(note === ""){
            let alerts = document.getElementById("alerts");
            alerts.style.display = "block";
            alerts.style.backgroundColor = "red";
            alerts.innerText = "Title is empty.";
            setTimeout(function(){
                alerts.innerText = "";
                alerts.style.backgroundColor = "white";
                alerts.style.display = "none";
            }, 3000)
        }
    }
    catch(error){
        console.error(error);
    }
    finally{
        let sendnote = document.getElementById("submit");
        sendnote.removeAttribute("disabled");
    }
}

export default NotesCreator;