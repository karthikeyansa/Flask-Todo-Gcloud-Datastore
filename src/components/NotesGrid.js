import React, { useEffect, useState } from "react";
import {API} from "../Apiservice";


function NotesGrid(props){
    const [notes, setNotes] = useState(props.Newnote);

    useEffect(()=>{
        async function notesChecker(){
            try{
                if(props.Newnote){
                    setNotes([props.Newnote, ...notes]);
                }
                else{
                    let response = await API.GetNotes();
                    if(response.success === true){
                        setNotes(response.result);
                    }
                }
            }
            catch(error){
                console.error(error);
            }
        }
        notesChecker();   
    }, [props.Newnote])

    const selectedNote = (note) => (event) =>{
        props.NoteSelected(note);
    } 

    const NoteRemover = (id) => (event) =>{
        async function deleteNote(){
            try{
                let response = await API.DeleteNote({id : id});
                if(response.success === true){
                    let displaycellRemoved = document.getElementById(id);
                    displaycellRemoved.style.display = "none";
                }
            }
            catch(error){
                console.error(error);
            }
        }
        deleteNote();
    }

    return(
        <React.Fragment>
            <div style={{display: "block",height: "100px"}}></div>
            <div className="displaygrid" id="displaygrid">
                {notes && notes.map((note)=>{
                    let completedTasks = 0;
                    let inCompletedTasks = 0;
                    return(
                    <div key={note.id}>
                        <div className="displaygridcell" id={note.id}>
                            <div style={{cursor: "pointer"}} onClick={selectedNote(note)}>
                                {note.content}
                            </div>
                                {note.tasks && note.tasks.forEach((task)=>{
                                    if (task.status === true){
                                        completedTasks += 1;
                                    }
                                    else if (task.status === false){
                                        inCompletedTasks += 1;
                                    }                     
                                })}
                            <div>
                                <span className="danger-btn" onClick={NoteRemover(note.id)}>üóëÔ∏è</span>&nbsp;
                                <span>Status: ‚ùåOngoing({inCompletedTasks}) <span style={{color: "lightgreen"}} >‚úî</span>Completed({completedTasks})</span>
                            </div>
                        </div>
                    </div>
                    )
                })}
            </div>
        </React.Fragment>
    )
}

export default NotesGrid;