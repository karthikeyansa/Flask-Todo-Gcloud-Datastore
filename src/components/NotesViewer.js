import React, {useEffect, useState} from "react";
import {API} from "../Apiservice";


window.completedTasks = 0;
window.ongoingTasks = 0;
window.note = {};

function NotesViewer(props){

    const [note] = useState(props.note);
    const [noteContent, setNoteContent] = useState("");
    const [newTask, setNewtask] = useState("");


    useEffect(()=>{
        countKeeper(window.completedTasks, window.ongoingTasks);
    },[])

    const countKeeper = (completedTasks, ongoingTasks) =>{
        note.tasks.forEach(task => {
            if(task.status === false){
                ongoingTasks += 1;
            }
            else if(task.status === true){
                completedTasks += 1;
            }
            window.ongoingTasks = ongoingTasks; 
            window.completedTasks = completedTasks;
            window.note = props.note;
        })
        let ongoingTasksCount = document.getElementById("incompleteTaskCount");
        if(ongoingTasksCount) ongoingTasksCount.innerText = `‚ùå Ongoing (${window.ongoingTasks})`;
        let completedTasksCount = document.getElementById("completeTaskCount");
        if(completedTasksCount) completedTasksCount.innerHTML = `<span style={{color:"lightgreen"}} >‚úî </span>Completed (${window.completedTasks})`;
    } 

    const expandTextarea = (event) =>{
        event.target.style.height="100px";
    }

    const noteContentarea = async(event) =>{
        let alerts = document.getElementById("alerts");
        if(noteContent !== ""){
            try{
                let response = await API.UpdateNoteandTaskcontent({note_id: note.id, content: noteContent});
                if(response.success === true){
                    window.note = response.result;
                    alerts.style.display = "block";
                    alerts.style.backgroundColor = "green";
                    alerts.innerText = "Note story updated";
                    setTimeout(()=>{
                        alerts.style.display = "none"
                    }, 1000)
                    setNoteContent("");
                }
            }
            catch(error){
                console.log(error);
            }
        }
        else if(newTask !== ""){
            try{
                let response = await API.UpdateNoteandTaskcontent({note_id: note.id, description: newTask});
                if(response.success === true){
                    window.note = response.result;
                    response.result.tasks.filter((task)=>{
                    if(task.description === newTask){
                        let incompleteTasks = document.getElementById("incompleteTasks");
                        
                        let taskStatusButton = document.createElement("span");
                        taskStatusButton.setAttribute("class", "modalIcon");
                        taskStatusButton.setAttribute("id", `span${task.task_id}`);
                        taskStatusButton.addEventListener("click", function(event){
                            checkStatus(note.id, task.task_id, event.target.innerText)
                        })
                        taskStatusButton.innerText = "‚ùå";
                        
                        let taskDescription = document.createElement("textarea");
                        taskDescription.setAttribute("class", "taskTextarea");
                        taskDescription.setAttribute("id", `descript${task.task_id}`);
                        taskDescription.innerHTML = newTask;
                        taskDescription.addEventListener("blur", function(){
                            noteTaskarea(note.id, task.task_id)
                        })  
                        taskDescription.addEventListener("focusin", function(){
                            taskDescription.style.height = "100px";
                            incompleteTasks.scrollTop = incompleteTasks.scrollHeight;
                        })
                                                    
                        let taskDeleteButton = document.createElement("span");
                        taskDeleteButton.setAttribute("id", `trash${task.task_id}`);
                        taskDeleteButton.setAttribute("class", "modalIcon");
                        taskDeleteButton.innerText = "üóëÔ∏è";
                        taskDeleteButton.addEventListener("click", function(){
                            taskRemover(note.id, task.task_id)
                        })
    
                        incompleteTasks.appendChild(taskStatusButton);
                        incompleteTasks.appendChild(taskDescription);
                        incompleteTasks.appendChild(taskDeleteButton);
                        incompleteTasks.scrollTop = incompleteTasks.scrollHeight;
    
                        let ongoingTasksCount = document.getElementById("incompleteTaskCount");
                        window.ongoingTasks = window.ongoingTasks + 1;
                        if(ongoingTasksCount) ongoingTasksCount.innerText = `‚ùå Ongoing (${window.ongoingTasks})`;
                    }})
                    setNewtask("");
                    let alerts = document.getElementById("alerts");
                    alerts.style.display = "block";
                    alerts.style.backgroundColor = "green";
                    alerts.innerText = "New task added.";
                    setTimeout(()=>{
                        alerts.style.display = "none"
                    }, 1000)
                }
            }
            catch(error){
                console.log(error);
            }
            
        }
    }

    const closeModal = () => {
        window.completedTasks = 0;
        window.ongoingTasks = 0;
        props.ModalClosed(window.note);

    }
    return(
        <React.Fragment>
            <br />
            <div className="alerts" id="alerts"></div>
            <button onClick={closeModal} className="modalCloseButton danger-btn">Close</button>
            <br />
            <div className="noteViewerGrid">
                <textarea className="noteContentTextarea" id="NoteContent" 
                          onBlur={noteContentarea} onChange={(e)=>setNoteContent(e.target.value)} defaultValue={note.content}></textarea>
                <div className="ongoingTasks"  id="incompleteTaskCount"></div>
                <div className="categoryTasks" id="incompleteTasks">
                    {note.tasks && note.tasks.filter((task)=>task.status === false).map((task)=>{
                        return(
                        <React.Fragment key={task.task_id}>
                            <span id={`span${task.task_id}`} style={{textAlign: "center"}} className="modalIcon" 
                                    onClick={(e) => {checkStatus(note.id, task.task_id, e.target.innerText)}}>‚ùå</span>
                            <textarea className="taskTextarea" 
                                        onBlur={()=>noteTaskarea(note.id, task.task_id)} 
                                        onSelect={expandTextarea} id={`descript${task.task_id}`} defaultValue={task.description}></textarea>
                            <span id={`trash${task.task_id}`} className="modalIcon" 
                                    onClick={()=>taskRemover(note.id, task.task_id)}>üóëÔ∏è</span>
                        </React.Fragment>
                        )
                    })}
                </div>
                <br/>
                <div className="newTaskTextarea" style={{height: "55px"}}>
                    <textarea className="newtexareaPadding" style={{height: "50px"}} 
                              placeholder="Add new task here." onChange={(e)=>{setNewtask(e.target.value)}} value={newTask}></textarea>
                    <span>&nbsp;</span>
                    <div className="newTaskButtonGrid">
                        <button className="newNoteButtons success-btn" onClick={noteContentarea}>Add</button>
                        <span style={{height:"0px"}}>&nbsp;</span>
                        <button className="newNoteButtons danger-btn" onClick={(e)=>setNewtask("")}>Cancel</button>
                    </div>
                </div>
                <br />
                <div className="completedTasks" id="completeTaskCount"></div>
                <div className="categoryTasks" id="completeTasks">
                    {note.tasks && note.tasks.filter((task)=>task.status === true).map((task)=>{
                        return(
                        <React.Fragment key={task.task_id}>
                            <span id={`span${task.task_id}`} className="modalIcon"
                                    onClick={(e) => {checkStatus(note.id, task.task_id, e.target.innerText)}}>‚úî</span>
                            <textarea className="taskTextarea" 
                                        onBlur={()=>noteTaskarea(note.id, task.task_id)} 
                                        onSelect={expandTextarea} id={`descript${task.task_id}`} defaultValue={task.description}></textarea>
                            <span id={`trash${task.task_id}`} className="modalIcon" 
                                    onClick={()=>taskRemover(note.id, task.task_id)}>üóëÔ∏è</span>
                        </React.Fragment>
                        )
                    })}
                </div>
            </div>
        </React.Fragment>
    )
}

async function checkStatus(note_id, task_id, status){
    var response = {};
    try{
        if(status === '‚úî'){
            response = await API.StatusUpdate({note_id: note_id, task_id: task_id, status: false})
        }
        else if(status === '‚ùå'){
            response = await API.StatusUpdate({note_id: note_id, task_id: task_id, status: true})
        }
    }
    catch(error){
        console.log(error);
    }
    if(response.success === true){
        window.note = response.result;
        let taskStatusButton = document.getElementById(`span${task_id}`);
        let taskdescript = document.getElementById(`descript${task_id}`);
        let trashButton = document.getElementById(`trash${task_id}`);
        taskStatusButton.remove();
        taskdescript.remove();
        trashButton.remove();
        if (taskStatusButton.innerText === "‚ùå") {
            taskStatusButton.innerText = "‚úî";
            let completedTasks = document.getElementById("completeTasks");
            completedTasks.appendChild(taskStatusButton);
            completedTasks.appendChild(taskdescript);
            completedTasks.appendChild(trashButton);
            completedTasks.scrollTop = completedTasks.scrollHeight;
            let completedTasksCount = document.getElementById("completeTaskCount");
            window.completedTasks = window.completedTasks + 1;
            if(completedTasksCount) completedTasksCount.innerHTML = `<span style={{color:"lightgreen"}} >‚úî </span>Completed (${window.completedTasks})`;
            // deducte by one
            let ongoingTasksCount = document.getElementById("incompleteTaskCount");
            window.ongoingTasks = window.ongoingTasks - 1;
            if(ongoingTasksCount) ongoingTasksCount.innerText = `‚ùå Ongoing (${window.ongoingTasks})`;
        
        }
        else if (taskStatusButton.innerText === "‚úî") {
            taskStatusButton.innerText = "‚ùå";
            let ongoingTasks = document.getElementById("incompleteTasks")
            ongoingTasks.appendChild(taskStatusButton);
            ongoingTasks.appendChild(taskdescript);
            ongoingTasks.appendChild(trashButton);
            ongoingTasks.scrollTop = ongoingTasks.scrollHeight;
            let ongoingTasksCount = document.getElementById("incompleteTaskCount");
            window.ongoingTasks = window.ongoingTasks + 1;
            if(ongoingTasksCount) ongoingTasksCount.innerText = `‚ùå Ongoing (${window.ongoingTasks})`;
            // deducte by one
            let completedTasksCount = document.getElementById("completeTaskCount");
            window.completedTasks = window.completedTasks - 1;
            if(completedTasksCount) completedTasksCount.innerHTML = `<span style={{color:"lightgreen"}} >‚úî </span>Completed (${window.completedTasks})`;
           
        }    
    }
}

async function taskRemover(note_id, task_id){
    let taskStatusButton = document.getElementById(`span${task_id}`);
    let taskdescript = document.getElementById(`descript${task_id}`);
    let taskDeleteButton = document.getElementById(`trash${task_id}`);
    let alerts = document.getElementById("alerts");
    let ongoingTasksCount = document.getElementById("incompleteTaskCount");
    let completedTasksCount = document.getElementById("completeTaskCount");
    try{
        let response = await API.RemoveTask({note_id: note_id, task_id: task_id});
        if(response.success === true){
            window.note = response.result;
            taskStatusButton.remove();
            taskdescript.remove();
            taskDeleteButton.remove();
            alerts.style.display = "block";
            alerts.style.backgroundColor = "green";
            alerts.innerText = "Task removed successfully";
            setTimeout(()=>{
                alerts.style.display = "none"
            }, 1000)
            if(taskStatusButton.innerText === "‚ùå"){
                window.ongoingTasks = window.ongoingTasks - 1;
                if(ongoingTasksCount) ongoingTasksCount.innerText = `‚ùå Ongoing (${window.ongoingTasks})`;
            }
            else if(taskStatusButton.innerText === "‚úî"){
                window.completedTasks = window.completedTasks - 1;
                if(completedTasksCount) completedTasksCount.innerHTML = `<span style={{color:"lightgreen"}} >‚úî </span>Completed (${window.completedTasks})`;
            }
        }
    }
    catch(error){
        console.log(error);
    }

}

async function noteTaskarea(note_id=0, task_id=0){
    let taskarea = document.getElementById(`descript${task_id}`);
    let alerts = document.getElementById("alerts");
    taskarea.style.height = "25px";
    if(taskarea.value !== ""){
        try{
            let response = await API.UpdateTaskDescription({note_id: note_id, task_id: task_id, description: taskarea.value});
            if(response.success === true){
                window.note = response.result;
                alerts.style.display = "block";
                alerts.style.backgroundColor = "green";
                alerts.innerText = "Task updated";
                setTimeout(()=>{
                    alerts.style.display = "none"
                }, 1000)
            }
        }
        catch(error){
            console.log(error);
        }
    }
}

export default NotesViewer;