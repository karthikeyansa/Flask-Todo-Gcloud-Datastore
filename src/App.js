import React, {useState} from "react";
import NotesCreator from "./components/NotesCreator";
import NotesGrid from "./components/NotesGrid";
import NotesViewer from "./components/NotesViewer";

function App() {
  const [note, setNote] = useState(null);
  const [notesGrid, setNotesgrid] = useState(true);
  const [notesCreator, setNotescreator] = useState(true);
  const [noteViewer, setNoteviewer] = useState(false);
  const [newNote, setNewNote] = useState(null);
  const [noteUpdate, setNoteupdate] = useState(null);
  const [allNotesData, setAllnotesData] = useState([]);

  const NoteSelected = (note) =>{
    setNote(note);
    setNotesgrid(false);
    setNotescreator(false);
    setNoteviewer(true);
    setNoteupdate(null);
  }
  const ModalClosed = (updatedNote) => {
    setNote(null);
    setNotesgrid(true);
    setNotescreator(true);
    setNoteviewer(false);
    setNewNote(null);
    setNoteupdate(updatedNote);
  }
  const newAddednote = (note) =>{
    setNewNote(note)
  }

  const notesData = (notes) => {
    setAllnotesData(notes);
  }

  return (
    <React.Fragment>
      {notesCreator && <NotesCreator newAddednote={newAddednote} />}
      {noteViewer && <NotesViewer note={note} ModalClosed={ModalClosed}/>}
      {notesGrid && <NotesGrid NoteSelected={NoteSelected} Newnote={newNote}
                               pushUpdatedNote={noteUpdate} notesData={notesData} allNotesData={allNotesData}/>}
    </React.Fragment>
  );
}

export default App;
