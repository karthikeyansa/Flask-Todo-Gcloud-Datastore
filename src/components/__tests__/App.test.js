import React from 'react';
import { mount, shallow } from 'enzyme';

import App from '../../App';
import NotesCreator from '../NotesCreator';
import NotesGrid from '../NotesGrid';
import NotesViewer from '../NotesViewer';
import renderer from 'react-test-renderer';

describe('App', () => {
  it('should render', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.exists()).toBeTruthy()
  })

  it("should render without crashing as per in snapshot App", ()=>{
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})

describe('App', () => {
  it('App component one note is selected state', () => {
    const holder = document.createElement("div");
    document.body.appendChild(holder);

    const note = {
      id: 123,
      content : "Hello world",
      tasks: [{
        task_id: 1,
        description: "task 1",
        status : true
      },
      {
        task_id : 2,
        description : "task 2",
        status: false
      }]
    }

    global.fetch = jest.fn(() => Promise.resolve({success: true}))
    const widgets = {
      note : note,
      notesGrid: false,
      notesCreator: false,
      noteViewer: true,
      noteUpdate: null
    }
    
    const wrapper = shallow(<App />);
  })
})