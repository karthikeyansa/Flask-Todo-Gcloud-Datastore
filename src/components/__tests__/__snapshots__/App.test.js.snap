// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing as per in snapshot App 1`] = `
Array [
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Welcome to Notes
  </h1>,
  <div
    className="options"
  >
    <span>
       
    </span>
    <span>
       
    </span>
    <button
      className="primary-btn"
      onClick={[Function]}
    >
      Profile
    </button>
    <span>
       
    </span>
    <button
      className="danger-btn"
      onClick={[Function]}
    >
      Logout
    </button>
    <span>
       
    </span>
  </div>,
  <div
    style={
      Object {
        "display": "block",
        "height": "25px",
      }
    }
  />,
  <div
    className="alerts"
    id="alerts"
  />,
  <div
    style={
      Object {
        "display": "block",
        "height": "40px",
      }
    }
  />,
  <h5
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Add here.
  </h5>,
  <div
    className="maingrid"
    id="maingrid"
  >
    <input
      className="input"
      id="content"
      name="content"
      onChange={[Function]}
      placeholder="Title"
      required={true}
      type="text"
      value=""
    />
     
    <div
      id="inputTask"
    >
      <textarea
        className="textarea"
        id="task1"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="List item"
        type="text"
        value=""
      />
      <span
        className="spanClose"
        id="span1"
        onClick={[Function]}
      >
        ❌
      </span>
    </div>
    <br />
    <button
      className="primary-btn"
      id="submitButton"
      onClick={[Function]}
      title="click to add to notes"
    >
      ➕Add to notes
    </button>
  </div>,
  <div
    className="displaygrid"
    id="displaygrid"
  />,
]
`;
